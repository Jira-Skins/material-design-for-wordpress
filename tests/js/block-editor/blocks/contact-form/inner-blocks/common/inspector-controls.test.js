/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import InspectorControls from '../../../../../../../assets/src/block-editor/blocks/contact-form/inner-blocks/common/components/inspector-controls';

/**
 * Render the component in a shallow mode using enzyme.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setupShallow = props => {
	return shallow( <InspectorControls { ...props } /> );
};

const baseProps = {
	attributes: {
		id: 'test-id',
		label: 'Test Label',
	},
};

describe( 'InspectorControls', () => {
	afterEach( () => {
		jest.clearAllMocks();
	} );

	it( 'matches snapshot', () => {
		const wrapper = setupShallow( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
