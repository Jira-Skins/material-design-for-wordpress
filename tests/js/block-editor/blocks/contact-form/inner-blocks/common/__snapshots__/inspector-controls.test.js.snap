// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InspectorControls matches snapshot 1`] = `
<IfBlockEditSelected(InspectorControlsFill)>
  <ForwardRef(PanelBody)
    initialOpen={true}
    title="Field Settings"
  >
    <TextControl
      label="ID"
      onChange={[Function]}
      value="test-id"
    />
    <TextControl
      label="Label"
      onChange={[Function]}
      value="Test Label"
    />
  </ForwardRef(PanelBody)>
</IfBlockEditSelected(InspectorControlsFill)>
`;
